src/riak_core_capability.erl

```diff
@@ -359,9 +363,23 @@ add_node(Node, [], State=#state{unknown=Unknown}) ->
                end,
     State2 = State#state{unknown=Unknown2},
     add_node_capabilities(Node, Capabilities, State2);
-add_node(Node, Capabilities, State) ->
+add_node2(Node, Capabilities, State) ->
     add_node_capabilities(Node, Capabilities, State).
 
+%% Remove supported capabilities for a node where it no longer appears in the
+%% rings capabilities for the node.
+%%
+%% This can occur when another node downgrades to a version where the capability
+%% is not declared. We must remove it from our local copy of its supported modes
+%% so that the default value can be used.
+filter_removed_caps(Node, Capabilities, State1) ->
+    {NodeSupported, _Removed} = lists:partition(
```

Comment:
Since you're just throwing away _Removed, please use `lists:filter` instead of `lists:partition`.